### PR TITLE
Null value deserialization into the JsonValue.NULL

### LINKED ISSUE #181: Deserialization from null to JsonValue
According to the current spec, mapping JSON document
```json
{
    "value": null
}
```
to the following Java class
```java
public class Stuff {
    public JsonValue value;
}
```
assigns `null` to the `value` in the class instead of `JsonValue.NULL`.

On the other hand, the following JSON document
```json
{
    "value": {
        "value2": null
    }
}
```
assigns a JSON object to the `value` and `JsonValue.NULL` is assigned to its `value2` property.

I believe `null` in JSON should be converted to `JsonValue.NULL` if the target type is `JsonValue`, in the same way it is converted specially to `Optional.empty()` when the target type is `Optional`.

Please see also eclipse-ee4j/yasson#293

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #181

### CODE REVIEW

File: tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java [hunk 1]
```diff
@@ -448,4 +448,24 @@ public void testJsonNumberMapping() {
...
+   * @test_Strategy: Assert that null is properly deserialized to the JsonValue.NULL
+   */
+  @Test
+  public void testNullDeserializedToJsonValueNull() {
+    JsonValueContainer unmarshalledValue = jsonb.fromJson("{ \"instance\" : null }", JsonValueContainer.class);
+    if (!JsonValue.NULL.equals(unmarshalledValue.getInstance())) {
+      fail("Failed to unmarshal null value to the JsonValue.NULL");
+    }
+    unmarshalledValue = jsonb.fromJson("{}", JsonValueContainer.class);
+    if (unmarshalledValue.getInstance() != null) {
```
Comment (lukasj) [inline tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java:466]: why not simple `Assert.assertNull(...)`?

Comment (lukasj) [reply tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java:466]: (same few lines above...)

Comment (Verdent) [reply tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java:466]: For some reason, these assertions are not used anywhere in the tests. I was thinking about doing it the same way for all the tests.

Comment (lukasj) [reply tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java:466]: I see no reason for not using them - they can have description of the cause/reason. (this wasn't the case for assertTrue/False due to bug in some older junit versions, where the description was ignored)

Comment (Verdent) [reply tck/src/main/java/jakarta/json/bind/tck/defaultmapping/jsonptypes/JSONPTypesMappingTest.java:466]: Alright. I will change that.

**Pull request merged**